src/callbacks_stage/antidiffusive_stage.jl

```diff
@@ -0,0 +1,73 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+
+"""
+    AntidiffusiveStage()
+
+Perform antidiffusive stage for IDP limiting.
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+struct AntidiffusiveStage end
+
+function (antidiffusive_stage!::AntidiffusiveStage)(u_ode, integrator, stage)
+
+  antidiffusive_stage!(u_ode, integrator.p, integrator.t, integrator.dt, integrator.p.solver.volume_integral)
+end
+
+(::AntidiffusiveStage)(u_ode, semi, t, dt, volume_integral::AbstractVolumeIntegral) = nothing
+
+function (antidiffusive_stage!::AntidiffusiveStage)(u_ode, semi, t, dt, volume_integral::VolumeIntegralShockCapturingSubcell)
+
+  @trixi_timeit timer() "antidiffusive_stage!" antidiffusive_stage!(u_ode, semi, t, dt, volume_integral.indicator)
+end
+
+function (antidiffusive_stage!::AntidiffusiveStage)(u_ode, semi, t, dt, indicator::IndicatorIDP)
+  mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
+
+  u = wrap_array(u_ode, mesh, equations, solver, cache)
+
+  @trixi_timeit timer() "alpha calculation" semi.solver.volume_integral.indicator(u, semi, solver, t, dt)
+
+  perform_IDP_correction(u, dt, mesh, equations, solver, cache)
+
+  return nothing
+end
+
+@inline function perform_IDP_correction(u, dt, mesh::TreeMesh2D, equations, dg, cache)
```

Comment:
We typically use `snake_case` for all variables and functions, so this should probably be
```suggestion
@inline function perform_idp_correction(u, dt, mesh::TreeMesh2D, equations, dg, cache)
```
(see also our short [style guide](https://trixi-framework.github.io/Trixi.jl/stable/styleguide/)). I know it sounds trite, but it really does help when reading the code if it is formatted consistently, thus it would be great if you could check this for other parts of your PR as well.